variables/util_test.go

```diff
@@ -0,0 +1,28 @@
+package variables
+
+import (
+	"testing"
+
+	. "github.com/stretchr/testify/assert"
+)
+
+func TestParseEscapedKeyValue(t *testing.T) {
+	var tests = []struct {
+		kv string
+		k  string
+		v  string
+		ok bool
+	}{
+		{"key=val", "key", "val", true},
+		{"with space=val with space", "with space", "val with space", true},
+		{"dudewheresmyvalue", "", "", false},
+		{`\==\\`, "=", `\`, true},
```

Comment:
We should add `key=value=something` too as a valid parse. I think we rely on something like this somewhere in our code (forgot where).